Add Rails 7.2, explain new EOL policy, and update EOL dates (#5635)
captn3m0/endoflife.date: products/ruby-on-rails.md

-   /ror
releasePolicyLink: https://guides.rubyonrails.org/maintenance_policy.html
changelogTemplate: https://github.com/rails/rails/releases/tag/v__LATEST__

> 

Reviewer: ```suggestion
```

:lipstick: Formatting

  -   git: https://github.com/rails/rails.git

> releases:

Reviewer: ```suggestion
# Some dates can be found on https://rubyonrails.org/maintenance.
releases:
```

For future maintainers.

  -   git: https://github.com/rails/rails.git

releases:
-   releaseCycle: "7.2"
    releaseDate: 2024-08-09
    eoas: 2025-08-09
    eol: 2026-08-09
    latest: "7.2.0"
    latestReleaseDate: 2024-08-09

-   releaseCycle: "7.1"
    releaseDate: 2023-10-05
    eoas: 2024-08-09

>     eol: 2025-10-01

Reviewer: ```suggestion
    eol: 2025-10-01 # see https://rubyonrails.org/maintenance
```

    latest: "7.1.3.4"
    latestReleaseDate: 2024-06-04

-   releaseCycle: "7.0"
    releaseDate: 2021-12-15
    eoas: 2023-10-15

>     eol: 2025-04-01

Reviewer: ```suggestion
    eol: 2025-04-01 # see https://rubyonrails.org/maintenance
```

    latest: "7.0.8.4"
    latestReleaseDate: 2024-06-04

-   releaseCycle: "6.1"
    releaseDate: 2020-12-09
    eoas: 2021-12-15

>     eol: 2024-10-01

Reviewer: ```suggestion
    eol: 2024-10-01 # see https://rubyonrails.org/maintenance
```

>[Ruby on Rails](https://rubyonrails.org/), or Rails, is a server-side web application framework
> written in Ruby.

> Each minor release cycle (7.2, 8.0, etc.) is officially supported based on a fixed, yearly duration: 1 year for bug fixes and 2 years for security fixes. For example, if a theoretical 1.1.0 is released on January 1, 2023, it will receive bug and security fixes until January 1, 2024 (1 year), and then receive security fixes only until January 1, 2025 (2 years). After that, it will reach its end-of-life.

Reviewer: Looks like this policy was update with 7.2: https://guides.rubyonrails.org/v7.2.0/maintenance_policy.html. Maintenance policy for those versions were still the old one: 7.0 (https://guides.rubyonrails.org/v7.0.8/maintenance_policy.html) and 7.1 (https://guides.rubyonrails.org/v7.1.3/maintenance_policy.html). Should we document this as well ? Is there a blog post or a public discussion somewhere that announce this change ?

Reviewer: Also found this link, which provides clear dates: https://rubyonrails.org/maintenance. Should we reference this link here ?

Author: @marcwrobel I was going by the discussion here https://github.com/rails/rails/pull/52471 and the official page here https://guides.rubyonrails.org/maintenance_policy.html . But it looks like the page you found has even better info. Thanks! I'll update the PR to reference that instead.

Author: @marcwrobel I updated this description to clarify that the policy is effective starting with Rails 7.2. I also changed the `releasePolicyLink` value to https://rubyonrails.org/maintenance. That automatically places this link on the rendered page below the description, like this:

> More information is available on the [Ruby on Rails website](https://rubyonrails.org/maintenance).

See c34d8288

> Starting with 7.2, each release cycle (7.2, 8.0, etc.) is officially supported based on a fixed, yearly duration: 1 year for bug fixes and 2 years for security fixes. For example, if a theoretical 1.1.0 is released on January 1, 2023, it will receive bug and security fixes until January 1, 2024 (1 year), and then receive security fixes only until January 1, 2025 (2 years). After that, it will reach its end-of-life.

Reviewer: ```suggestion
[Starting with 7.2](https://github.com/rails/rails/pull/52471#issuecomment-2271508281), each minor release (7.2, 8.0, etc.) is officially supported based on a fixed, yearly duration: 1 year for bug fixes and 2 years for security fixes. For example, if a theoretical 1.1.0 is released on January 1, 2023, it will receive bug and security fixes until January 1, 2024 (1 year), and then receive security fixes only until January 1, 2025 (2 years). After that, it will reach its end-of-life.
```

Minor update, add link to source.
